File: ui/src/app/model/projeto-ata-reuniao.ts
import { FormArray, FormControl, FormGroup, Validators } from '@angular/forms';

import { Projeto } from './projeto';
import { ProjetoFase } from './projeto-fase';
import { Usuario } from './usuario';
import { ProjetoAtaReuniaoParticipante } from './projeto-ata-reuniao-participante';

export class ProjetoAtaReuniao {

  id: number | undefined;
  codigo: number | undefined;
  projeto: Projeto | undefined;
  projetoFase: ProjetoFase | undefined; //
  dataHora: string | undefined;
  ata: string | undefined;
  enviaEmail: boolean | undefined;
  tituloEmail: string | undefined;
  corpoEmail: string | undefined;
  projetoAtaReuniaoParticipantes: ProjetoAtaReuniaoParticipante [] | undefined;

  constructor() {
    this.enviaEmail = false;
    this.projetoAtaReuniaoParticipantes = [];
  }

  criarFormulario(projetoAtaReuniao: ProjetoAtaReuniao) {

    return new FormGroup({
      id: new FormControl(projetoAtaReuniao.id),
      codigo: new FormControl(projetoAtaReuniao.codigo, Validators.required),
      projeto: new FormControl(projetoAtaReuniao.projeto, Validators.required),
      projetoFase: new FormControl(projetoAtaReuniao.projetoFase),
      dataHora: new FormControl(projetoAtaReuniao.dataHora, Validators.required),
      ata: new FormControl(projetoAtaReuniao.ata, Validators.required),
      enviaEmail: new FormControl(projetoAtaReuniao.enviaEmail),
      tituloEmail: new FormControl(projetoAtaReuniao.tituloEmail),
      corpoEmail: new FormControl(projetoAtaReuniao.corpoEmail),
      projetoAtaReuniaoParticipantes: new FormArray(projetoAtaReuniao!.projetoAtaReuniaoParticipantes!.map(resp => ProjetoAtaReuniaoParticipante.criarFormulario(resp)))
    });
  }
  
}

File: api/src/main/java/com/example/demo/model/ProjetoAtaReuniao.java
package com.example.demo.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@Table(name = "projeto_ata_reuniao")
public class ProjetoAtaReuniao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "codigo")
    private Integer codigo;

    @ManyToOne
    @JoinColumn(name = "projeto_id")
    private Projeto projeto;

    @ManyToOne
    @JoinColumn(name = "projeto_fase_id")
    private ProjetoFase projetoFase;

    @Column(name = "data_hora")
    private LocalDateTime dataHora;

    @Column(name = "ata")
    private String ata;

    @Column(name = "envia_email")
    private boolean enviaEmail;

    @Column(name = "titulo_email")
    private String tituloEmail;

    @Column(name = "corpo_email")
    private String corpoEmail;

    @ElementCollection
    @CollectionTable(name = "projeto_ata_reuniao_participante", joinColumns = {@JoinColumn(name = "projeto_ata_reuniao_id")})
    private List<ProjetoAtaReuniaoParticipante> projetoAtaReuniaoParticipantes = new ArrayList<>();

}
